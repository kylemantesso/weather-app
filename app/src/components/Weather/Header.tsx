import React, { useContext } from "react";
import moment from "moment";

import { Box, Heading, ResponsiveContext, Text } from "grommet";
import { FormatSwitch } from "./FormatSwitch";
import { FormatType } from "../../context/AppContext";

export const Header: React.FC<{
  location: string;
  formatType: FormatType;
  current: string;
}> = ({ location, formatType, current }) => {
  const size = useContext(ResponsiveContext);
  return (
    <Box direction="row-responsive" fill justify="between" gap="small">
      <Box direction="column" gap="xxsmall">
        <Heading size="large" margin="none">
          {location}
        </Heading>
        <Text>{moment().format("dddd, MMMM Do")}</Text>
        <Text>{current}</Text>
      </Box>
      <Box justify="start" align={size === "small" ? "start" : "end"}>
        <FormatSwitch formatType={formatType} />
      </Box>
    </Box>
  );
};
