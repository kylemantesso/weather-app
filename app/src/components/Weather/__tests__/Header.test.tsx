import React from "react";
import renderer from "react-test-renderer";
import { Header } from "../Header";

describe("Header", () => {
  it("matches snapshot", () => {
    const component = renderer.create(
      <Header location={"Melbourne"} formatType={"C"} current={"Clear"} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
