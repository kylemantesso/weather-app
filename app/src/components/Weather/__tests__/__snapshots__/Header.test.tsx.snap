// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 cCFlWA"
>
  <div
    className="StyledBox-sc-13pk1d4-0 bFwQzJ"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 kXGHwd"
      size="large"
    >
      Melbourne
    </h1>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 fLIXsc"
    />
    <span
      className="StyledText-sc-1sadyjn-0 hUokoe"
    >
      Thursday, August 13th
    </span>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 fLIXsc"
    />
    <span
      className="StyledText-sc-1sadyjn-0 hUokoe"
    >
      Clear
    </span>
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 cPQeYT"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 ANZcC"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 bIdaiR"
    >
      <button
        className="StyledButton-sc-323bzc-0 kphDHx"
        id="c"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        °C
      </button>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
      />
      <button
        className="StyledButton-sc-323bzc-0 kFNfEv"
        id="f"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        °F
      </button>
    </div>
  </div>
</div>
`;
